proto/workload.proto

```diff
@@ -17,6 +17,55 @@ syntax = "proto3";
 package istio.workload;
 option go_package="pkg/workloadapi";
 
+// Address represents a unique address, keyed by IP address and network.
+// An Address can represent a Service or an individual workload.
+// Address is intended to be able to be looked up on-demand, allowing a workload
+// to answer a question like "what is this IP address", similar to a reverse DNS lookup.
+// Keyed by network/IP address. Example: "default/1.2.3.4".
+//
+// In some cases, we do not know the IP address of a Workload. For instance, we may simply know
+// that there is a workload behind a gateway, and rely on the gateway to handle the rest.
+// In this case, the key format will be "network/resource-uid". The resource can be a Pod, WorkloadEntry, etc.
+// These resources cannot be looked up on-demand.
+//
+// In some cases, we do not know the IP address of a Service. These services cannot be used for matching
+// outbound traffic, as we only have L4 attributes to route based on. However,
+// they can be used for Gateways.
+// In this case, the key format will be "network/hostname".
+// These resources cannot be looked up on-demand.
+message Address {
+  oneof type {
+    // Workload represents an individual workload.
+    // This could be a single Pod, a VM instance, etc.
+    Workload workload = 1;
+    // Service represents a service - a group of workloads that can be accessed together.
+    Service service = 2;
+  }
+}
+
+// Service represents a service - a group of workloads that can be accessed together.
+message Service {
+  // Name represents the name for the service.
+  // For Kubernetes, this is the Service name.
+  string name = 1;
+  // Namespace represents the namespace for the service.
+  string namespace = 2;
+  // Hostname represents the FQDN of the service.
+  // For Kubernetes, this would be <name>.<namespace>.svc.<cluster domain>
+  string hostname = 3;
+  // Address represents the IPv4/IPv6 address for the service.
+  // This should be globally unique.
+  // This should not have a port number.
+  // TODO: Add network as discriminator
+  // TODO: Add dual stack support
+  bytes address = 4;
```

Comment:
which ip address should this be when it is in multi clusters

Reply:
I believe per the design doc this will become `NetworkAddress` once I implement that (before this merges and I ask for more reviews)

Reply:
the api has been updated, please take a look again